frontend/appflowy_flutter/lib/user/presentation/screens/sign_in_screen/mobile_sign_in_screen.dart

```diff
@@ -18,28 +19,43 @@ class MobileSignInScreen extends StatelessWidget {
   Widget build(BuildContext context) {
     const double spacing = 16;
     final colorScheme = Theme.of(context).colorScheme;
-    return Scaffold(
-      body: Padding(
-        padding: const EdgeInsets.symmetric(vertical: 50, horizontal: 40),
-        child: Column(
-          children: [
-            const Spacer(flex: 4),
-            _buildLogo(),
-            const VSpace(spacing * 2),
-            _buildWelcomeText(),
-            _buildAppNameText(colorScheme),
-            const VSpace(spacing * 2),
-            const SignInWithMagicLinkButtons(),
-            const VSpace(spacing),
-            if (isAuthEnabled) _buildThirdPartySignInButtons(colorScheme),
-            const VSpace(spacing),
-            const SignInAnonymousButtonV2(),
-            const VSpace(spacing),
-            _buildSettingsButton(context),
-            if (!isAuthEnabled) const Spacer(flex: 2),
-          ],
-        ),
-      ),
+    return BlocBuilder<SignInBloc, SignInState>(
+      builder: (context, state) {
+        return Scaffold(
+          body: Padding(
+            padding: const EdgeInsets.symmetric(vertical: 50, horizontal: 40),
+            child: Column(
+              children: [
+                const Spacer(flex: 4),
+                _buildLogo(),
+                const VSpace(spacing * 2),
+                _buildWelcomeText(),
+                _buildAppNameText(colorScheme),
+                const VSpace(spacing * 2),
+                const SignInWithMagicLinkButtons(),
+                const VSpace(spacing),
+                if (isAuthEnabled) _buildThirdPartySignInButtons(colorScheme),
+                const VSpace(spacing),
+                const SignInAnonymousButtonV2(),
+                const VSpace(spacing),
+                SwitchSignInSignUpButton(
+                  onTap: () {
+                    final type = state.loginType == LoginType.signIn
+                        ? LoginType.signUp
+                        : LoginType.signIn;
+                    context.read<SignInBloc>().add(
+                          SignInEvent.switchLoginType(type),
+                        );
+                  },
```

Comment:
```suggestion
                  onTap: () => context.read<SignInBloc>().add(
                          const SignInEvent.toggleLoginType(),
                        );
```
_personal preference_